app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt

```diff
@@ -0,0 +1,296 @@
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_submitTimeError_extraSpaces_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError(" 1   / 2 ")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_submitTimeError_tooLong_returnsNumberTooLong() {
+    val error = stringToFractionParser.getSubmitTimeError("0123456789")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error)
+      .isEqualTo("None of the numbers in the fraction should have more than 7 digits.")
+  }
+
+  @Test
+  fun testParser_submitTimeError_atLengthLimit_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError("1234567/1234567")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_submitTimeError_returnsDivisionByZero() {
+    val error = stringToFractionParser.getSubmitTimeError("123/0")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please do not put 0 in the denominator")
+  }
+
+  @Test
+  fun testParser_submitTimeError_ambiguousSpacing_returnsInvalidFormat() {
+    val error = stringToFractionParser.getSubmitTimeError("1 2 3/4")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_submitTimeError_emptyString_returnsInvalidFormat() {
+    val error = stringToFractionParser.getSubmitTimeError("")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_noNumerator_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("/3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_severalSlashes_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("1/3/8")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_severalDashes_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("-1/-3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_parseFractionFromString_incorrectFraction_throwsException() {
+    val exception = assertThrows(IllegalArgumentException::class) {
+      stringToFractionParser.parseFractionFromString("abc")
+    }
+    assertThat(exception)
+      .hasMessageThat()
+      .contains("Incorrectly formatted fraction: abc")
+  }
+
+  @Test
+  fun testParser_parseFraction_incorrectFraction_returnsNull() {
```

Comment:
I suggest updating this & the previous test's name to be clearer on which specific incorrect fraction is being tested.

Also, I think it might be worth adding tests for other failure cases when calling the parser (to make sure they have parity with the error functions).

Reply:
Done, updated test names and added more tests.